full-service/src/service/gift_code.rs

```diff
@@ -53,6 +54,8 @@ use rand::Rng;
 use serde::{Deserialize, Serialize};
 use std::{convert::TryFrom, fmt, iter::empty, str::FromStr, sync::atomic::Ordering};
 
+use super::ledger::LedgerServiceError;
```

Comment:
Nit: import using `crate::` like done above

Reply:
ah, auto import likes to default to using super for some reason